File: CPP/PROGRAM/Keyboard.cpp
// KELAS KEYBOARD
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Keyboard : public Komponen {
    private:
        string tipeKeyboard; // Mechanical, Membrane, dll
        string layout; // QWERTY, DVORAK, dll
        string backlit; // Ya/Tidak

    public:
        Keyboard() {}
        
        Keyboard(string merk, string nama, string produsen, string tipeKeyboard, string layout, string backlit) 
            : Komponen(merk, nama, produsen) {
            this->tipeKeyboard = tipeKeyboard;
            this->layout = layout;
            this->backlit = backlit;
        }
        
        void setTipeKeyboard(string tipeKeyboard) {
            this->tipeKeyboard = tipeKeyboard;
        }
        
        void setLayout(string layout) {
            this->layout = layout;
        }
        
        void setBacklit(string backlit) {
            this->backlit = backlit;
        }
        
        string getTipeKeyboard() {
            return tipeKeyboard;
        }
        
        string getLayout() {
            return layout;
        }
        
        string getBacklit() {
            return backlit;
        }
        
        void tampilkanInfo() {
            Komponen::tampilkanInfo();
            cout << "Tipe Keyboard: " << tipeKeyboard << endl;
            cout << "Layout: " << layout << endl;
            cout << "Backlit: " << backlit << endl;
        }
    
    ~Keyboard() {}
};

File: PYTHON/PROGRAM/Keyboard.py
# KELAS KEYBOARD
from Komponen import Komponen

class Keyboard(Komponen):
    def __init__(self, merk="", nama="", produsen="", tipe_keyboard="", layout="", backlit=""):
        Komponen.__init__(self, merk, nama, produsen)
        self.__tipe_keyboard = tipe_keyboard
        self.__layout = layout
        self.__backlit = backlit
    
    def set_tipe_keyboard(self, tipe_keyboard):
        self.__tipe_keyboard = tipe_keyboard
    
    def set_layout(self, layout):
        self.__layout = layout
    
    def set_backlit(self, backlit):
        self.__backlit = backlit
    
    def get_tipe_keyboard(self):
        return self.__tipe_keyboard
    
    def get_layout(self):
        return self.__layout
    
    def get_backlit(self):
        return self.__backlit
    
    def tampilkan_info(self):
        Komponen.tampilkan_info(self)
        print("Tipe Keyboard:", self.__tipe_keyboard)
        print("Layout:", self.__layout)
        print("Backlit:", self.__backlit)